### PR TITLE
chore: convert terminate_working_block macro to fn

### PR DESCRIPTION
**Pull request by nilslice**
While building up some more familiarity with #75, this was easier to follow w/ language server help. I think this function is pretty well covered in other tests, which would have caught issues, but that's an area I could look into more.

PR branch targets and is based on `75-frame-dependencies`. 

UPDATE: forgot to test with `--all-features` locally.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +9,12 @@ It serves three purposes:
 3. Construct a dependency graph among program instructions
 
 It should be considered unstable until the release of v1.0.
+
+
+## Testing
+
+When testing this crate, you should run with the `--all-features` flag to ensure all tests are executed.
```
Comment (kalzoo) [inline README.md:16]: 👍 true

**Review by kalzoo — APPROVED** [review]
Good change, thank you 🙏

File: README.md (hunk 1 above)
Comment (genos) [reply README.md:16]: 👏

File: src/program/graph.rs [hunk 2]
```diff
@@ -452,33 +452,36 @@ pub struct ScheduledProgram {
...
+fn terminate_working_block(
+    terminator: Option<BlockTerminator>,
+    working_instructions: &mut Vec<Instruction>,
+    blocks: &mut IndexMap<String, InstructionBlock>,
+    working_label: &mut Option<String>,
+    program: &Program,
+    instruction_index: Option<usize>,
+) -> ScheduleResult<()> {
+    // If this "block" has no instructions and no terminator, it's not worth storing - skip it
+    if working_instructions.is_empty() && terminator.is_none() && working_label.is_none() {
```
Comment (genos) [inline src/program/graph.rs:465]: Since you assign `*working_label = None` in both cases, you could pull that outside the conditional and have a single arm:
```rust 
if !working_instructions.is_empty() || !terminator.is_none() || !working_label.is_none() {
    …
}
*working_label = None;
Ok(())
```

File: src/program/graph.rs [hunk 3]
```diff
@@ -452,33 +452,36 @@ pub struct ScheduledProgram {
...
-                    variant: ScheduleErrorVariant::DuplicateLabel,
-                }), // Duplicate label
-                None => Ok(()),
-            }?;
-            $working_instructions = vec![];
-            $working_label = None
-        }
-        Ok(())
-    }};
+#[allow(unused_assignments)]
```
Comment (genos) [inline src/program/graph.rs:455]: Not too troubled, just curious: why is this necessary? Because of reassigning `working_label` at the end?

File: src/program/graph.rs [hunk 4]
```diff
@@ -452,33 +452,36 @@ pub struct ScheduledProgram {
...
+        let label = working_label
+            .clone()
+            .unwrap_or_else(|| ScheduledProgram::generate_autoincremented_label(blocks));
+
+        match blocks.insert(label.clone(), block) {
+            Some(_) => Err(ScheduleError {
+                instruction_index,
+                instruction: Instruction::Label(Label(label)),
+                variant: ScheduleErrorVariant::DuplicateLabel,
+            }), // Duplicate label
```
Comment (genos) [inline src/program/graph.rs:478]: With `variant: ScheduleErrorVariant::DuplicateLabel` here, the comment seems a bit superfluous.

**Review by genos — COMMENTED** [review]
Nice! A question for my own edification, some few ignorable suggestions, but LGTM.

File: src/program/graph.rs (hunk 2 above)
Comment (nilslice) [reply src/program/graph.rs:466]: good eye -- updated to early return!

File: src/program/graph.rs (hunk 3 above)
Comment (nilslice) [reply src/program/graph.rs]: Yep, no longer needed though after the branching change

File: src/program/graph.rs (hunk 4 above)
Comment (nilslice) [reply src/program/graph.rs]: removed! good catch ✅

File: src/program/graph.rs [hunk 5]
```diff
@@ -452,33 +452,40 @@ pub struct ScheduledProgram {
...
-                None => Ok(()),
-            }?;
-            $working_instructions = vec![];
-            $working_label = None
-        }
-        Ok(())
-    }};
+/// Builds an [`InstructionBlock`] from provided instructions, terminator, and program, then tracks
+/// the block and its label, and resets the instruction list and label for a future block to use.
+fn terminate_working_block(
```
Comment (nilslice) [inline src/program/graph.rs:457]: @kalzoo - feedback on depth/correctness of this comment here would be great, thanks!

File: src/program/graph.rs [hunk 6]
```diff
@@ -452,33 +452,40 @@ pub struct ScheduledProgram {
...
+    terminator: Option<BlockTerminator>,
+    working_instructions: &mut Vec<Instruction>,
+    blocks: &mut IndexMap<String, InstructionBlock>,
+    working_label: &mut Option<String>,
+    program: &Program,
+    instruction_index: Option<usize>,
+) -> ScheduleResult<()> {
+    // If this "block" has no instructions and no terminator, it's not worth storing - skip it
+    if working_instructions.is_empty() && terminator.is_none() && working_label.is_none() {
+        *working_label = None;
```
Comment (nilslice) [inline src/program/graph.rs:467]: Ah this can be removed, since we just check that `working_label.is_none()` the line above...

File: src/program/graph.rs [hunk 7]
```diff
@@ -365,33 +452,40 @@ pub struct ScheduledProgram {
...
-                None => Ok(()),
-            }?;
-            $working_instructions = vec![];
-            $working_label = None
-        }
-        Ok(())
-    }};
+/// Builds an [`InstructionBlock`] from provided instructions, terminator, and program, then tracks
+/// the block and its label, and resets the instruction list and label for a future block to use.
+fn terminate_working_block(
```
Comment (kalzoo) [inline src/program/graph.rs:457]:
```suggestion
/// Builds an [`InstructionBlock`] from provided instructions, terminator, and program, then tracks
/// the block and its label, and resets the instruction list and label for a future block to use.
///
/// The "working block" is that being traversed within the program, accumulating instructions located
/// between control instructions (such as `LABEL` and 'JUMP`). When such a control instruction is reached,
/// this function performs the work to close out and store the instructions in the current "working block"
/// and then reset the state to prepare for the next block.
fn terminate_working_block(
```

File: src/program/graph.rs (hunk 5 above)
Comment (kalzoo) [reply src/program/graph.rs:457]: made a suggestion to add some context on the `working_block` term, but otherwise lgtm 👍
